File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        res = []
        q = []
        q.append(root)
        level = 0
        while q:
            res.extend([0])
            for _ in range(len(q)):
                node = q.pop(0)
                last_seen = node.val

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                
                res[level] = last_seen
            level += 1
        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
private:
    vector<vector<int>> res;
    void dfs(TreeNode* root, int level){
        if (!root) return;
        if (level >= res.size()){
            res.push_back({});
        }

        res[level].push_back(root->val);
        dfs(root->left, level+1);
        dfs(root->right, level+1);
    }

public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        dfs(root, 0);
        return res;
    }
};
